src/datasets/utils/version.py

```diff
@@ -35,10 +36,10 @@ class Version:
     """
 
     version_str: str
-    description: str = None
-    major: str = None
-    minor: str = None
-    patch: str = None
+    description: Optional[str] = None
+    major: Optional[str] = None
+    minor: Optional[str] = None
+    patch: Optional[str] = None
```

Comment:
```suggestion
    major: str = None
    minor: str = None
    patch: str = None
```
These three fields are strings. They're set during post_init

Reply:
Checked the souce and it seems these three fields are ints (post_init calls _str_to_version which returns the 3-tuple of ints; the allow_wildcard param of _str_to_version is False by default so '*' is not allowed). I think the correct typing would look like this:

```python
major: int = field(init=False)
minor: int = field(init=False)
patch: int = field(init=False)
```

field(init=False) excludes them from the `__init__` (only version_str is required due to the steps performed in the post_init that are described above).

Another improvement would be to annotate the version_str field with `InitVar[str]`. This would make post_init's implementation cleaner:
```python
def __post_init__(self, version_str):
    self.major, self.minor, self.patch = _str_to_version(version_str) # no longer requires self.version_str
``` 
but with this change dataclasses.fields(cls) can no longer be used in Version.from_dict since it doesn't include info about the class InitVars. So implementation of Version.from_dict would be as follows:
```python
@classmethod
def from_dict(cls, dic):
    field_names = set(f.name for f in cls.__dataclass_fields__.values() if f.init)
    return cls(**{k: v for k, v in dic.items() if k in field_names})
```

Reply:
Thanks for the details.
Let's keep it this way and discuss this in another PR